src/main/java/by/mifort/automation/hr/dev/service/impl/CandidateServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package by.mifort.automation.hr.dev.service.impl;
+
+import by.mifort.automation.hr.dev.dto.CandidateDto;
+import by.mifort.automation.hr.dev.dto.FilterDto;
+import by.mifort.automation.hr.dev.entity.Candidate;
+import by.mifort.automation.hr.dev.entity.CandidateAttributes;
+import by.mifort.automation.hr.dev.entity.Keyword;
+import by.mifort.automation.hr.dev.repository.CandidateRepository;
+import by.mifort.automation.hr.dev.service.CandidateService;
+import by.mifort.automation.hr.dev.util.MappingDtoComponentConverter;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class CandidateServiceImpl implements CandidateService {
+
+    private final CandidateRepository candidateRepository;
+    private final MappingDtoComponentConverter converter;
+
+    @Autowired
+    public CandidateServiceImpl(CandidateRepository candidateRepository, MappingDtoComponentConverter converter) {
+        this.candidateRepository = candidateRepository;
+        this.converter = converter;
+    }
+
+    @Override
+    public List<CandidateDto> getAllPeople(FilterDto filterDto, String keyword) {
+        List<Candidate> candidateList;
+        Integer page = filterDto.getPage();
+        Integer amount = filterDto.getAmount();
+        if (page == null || amount == null || page <= 0 || amount <= 0) {
```

Comment:
we should check it in controller

Reply:
Why? All validation and logic we need to put in Service Layer. I want to create in future interface with validation

Reply:
it is validation of input param and you use here Http specific objects `ResponseStatusException(HttpStatus.BAD_REQUEST`